taker/src/routes.rs

```diff
@@ -78,6 +108,23 @@ pub async fn feed(
         tokio::time::interval(std::time::Duration::from_secs(HEARTBEAT_INTERVAL_SECS));
 
     EventStream! {
+        let mut rx_offers = match rx_offers {
+            None => {
+                // This should never happen as long as we instantiate `rx_offers` for each ContractSymbol variant
+                tracing::error!("No offer receiver for {symbol} found");
+                return ();
+            }
+            Some(rx_offers) => rx_offers.clone(),
+        };
+        let mut rx_quote = match rx_quote {
+            None => {
+                // This should never happen as long as we instantiate `rx_quote` for each ContractSymbol variant
+                tracing::error!("No quote receiver for {symbol} found");
+                return ();
+            }
+            Some(rx_quote) => rx_quote.clone(),
+        };
```

Comment:
🔧 I would opt to have dedicated `rx_{btcusd,ethusd}_offers` receivers. We can avoid this annoying boilerplate and I don't think we will ever miss the hash-map's dynamism.

Reply:
The advantage of the hash-map is that we should be able to just add a new contract symbol by just adding a new enum variant

Reply:
Done. I think it turned out nice.